### DISCUSSION TITLE
After using subset_fonts, some Greece letters are not displayed

### DISCUSSION THREAD

**Discussion by izerui** [pymupdf/PyMuPDF#3883] [Q&A] [answered]
def test_subfont0():
    doc = pymupdf.open()
    page = doc.new_page(width=1024, height=768)
    page.insert_htmlbox(
        rect=pymupdf.Rect(280, 50, 480, 100),
        text=f'<span style="font-size:18px;font-weight:bold;word-break:break-all;font-family:\'helv\';">Α、Β、Γ、Δ、Ε、Ζ、Η、Θ、Ι、Κ、Λ、Μ、Ν、Ξ、Ο、Π、Ρ、Σ、Τ、Υ、Φ、Χ、Ψ、Ω。α、β、γ、δ、ε、ζ、η、θ、ι、κ、λ、μ、ν、ξ、ο、π、ρ、ς、τ、υ、φ、χ、ψ、ω。</span>'
    )
    doc.subset_fonts()
    doc.ez_save('subseted.pdf')

**Comment by JorjMcKie** [comment, marked as answer]
If you do things like the above, you leave your fate in the hands of MuPDF's automatic font selection algorithm. If I stood in your shoes I would exert **_some_** influence and specify at least a font family to choose from, e.g. "sans-serif".
Other than that, why did you take "、" instead of a normal comma? That character, has the Unicode 0x3001 (12289, 'ideographiccomma') and is far outside the Extended Latin character set.
The same is true for "。" (0x3002, 12290, 'ideographicperiod'). Why not a normal colon?
Both characters only **_appear_** to be followed by a space, but there is no space, so crazy line breaks (in the middle of a word) can easily occur.

So with this text, you will end up with 3 fonts automatically identified by MuPDF. One of them, because of the 2 ideographic punctuations, is **_very large:_**
```python
(6, 'otf', 'Type0', 'Noto Serif Regular', 'F0', 'Identity-H', 24)
(12, 'cid', 'Type0', 'Charis SIL Regular', 'F1', 'Identity-H', 24)
(18, 'otf', 'Type0', 'Source Han Serif TC Regular', 'F2', 'Identity-H', 24)
```
If you do `page.insert_htmlbox(..., css="* {font-family: sans-serif;}")`, it will be just the following two:
```python
(6, 'cid', 'Type0', 'Nimbus Sans Regular', 'F0', 'Identity-H', 18)
(12, 'otf', 'Type0', 'Source Han Serif TC Regular', 'F1', 'Identity-H', 18)
```
If you replace the punctuation by normal comma and colon, only one front will be used (Nimbus Sans Regular).
Other than the above, there indeed is a problem with MuPDF's font subset algorithm, which I will report in their issue management.
You can use `doc.subset_fonts(fallback=True)` which will use the mechanism of the `fontTools` package.

**Reply by izerui** [reply]
`doc.subset_fonts(fallback=True)` it`s ok.

Thank you very much!!!

The only drawback is that the execution speed of subset_fonts is 4 to 5 times slower.
